connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerMetricsGroupTest.java

```diff
@@ -0,0 +1,164 @@
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.connect.runtime;
+
+import org.apache.kafka.common.metrics.Sensor;
+import org.apache.kafka.connect.util.ConnectorTaskId;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.powermock.api.easymock.PowerMock;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import static org.easymock.EasyMock.eq;
+import static org.powermock.api.easymock.PowerMock.createStrictMock;
+import static org.powermock.api.easymock.PowerMock.expectLastCall;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({Sensor.class})
+public class WorkerMetricsGroupTest {
+    private final String connector = "org.FakeConnector";
+    private final ConnectorTaskId task = new ConnectorTaskId(connector, 0);
+    private final RuntimeException exception = new RuntimeException();
+
+    @Test
+    public void testConnectorStartupRecordedMetrics() {
+        final WorkerMetricsGroup mockWorkerMetricsGroup = createStrictMock(WorkerMetricsGroup.class);
+        final ConnectorStatus.Listener delegate = createStrictMock(ConnectorStatus.Listener.class);
+        final WorkerMetricsGroup.ConnectorStatusListener connectorListener = mockWorkerMetricsGroup.new ConnectorStatusListener(delegate);
+
+        delegate.onStartup(connector);
+        expectLastCall();
+
+        mockWorkerMetricsGroup.recordConnectorStartupSuccess();
+        expectLastCall();
+
+        PowerMock.replayAll();
+
+        connectorListener.onStartup(connector);
+
+        PowerMock.verifyAll();
+    }
+
+    @Test
+    public void testConnectorFailureAfterStartupRecordedMetrics() {
+        final WorkerMetricsGroup mockWorkerMetricsGroup = createStrictMock(WorkerMetricsGroup.class);
+        final ConnectorStatus.Listener delegate = createStrictMock(ConnectorStatus.Listener.class);
+        final WorkerMetricsGroup.ConnectorStatusListener connectorListener = mockWorkerMetricsGroup.new ConnectorStatusListener(delegate);
+
+        delegate.onStartup(eq(connector));
+        expectLastCall();
+
+        delegate.onFailure(eq(connector), eq(exception));
+        expectLastCall();
+
+        mockWorkerMetricsGroup.recordConnectorStartupSuccess();
```

Comment:
Nit: I think it might make more sense if the expectations are set in chronological order instead of grouping by which mocked instance is having expectations set. So in this case, this line would be moved after the expectation for `delegate::onStartup` and before the one for `delegate::onFailure`. But not a big deal, if you think this is more readable feel free to leave as-is.

Reply:
No argument from me. I'll change it.